### PR TITLE
Moby-engine installation

### PR DESCRIPTION
**Pull request by ds-ms**
This PR replaces docker-ce installation with ms-moby installation.

### CODE REVIEW

File: images/linux/scripts/installers/docker-moby.sh [hunk 1]
```diff
@@ -19,10 +17,10 @@ if ! IsInstalled $docker_package; then
     apt-get remove -y docker docker-engine docker.io
```
Comment (bansalaseem) [inline images/linux/scripts/installers/docker-moby.sh]: Should we remove the moby engine here?

File: images/linux/scripts/installers/docker.sh [hunk 2]
```diff
@@ -21,10 +21,10 @@ if ! IsInstalled $DOCKER_PACKAGE; then
     apt-get remove -y docker docker-engine docker.io
```
Comment (bansalaseem) [inline images/linux/scripts/installers/docker.sh]: Same as above

File: images/linux/scripts/installers/docker-moby.sh (hunk 1 above)
Comment (ds-ms) [reply images/linux/scripts/installers/docker-moby.sh]: @bansalaseem Done.

File: images/linux/scripts/installers/docker.sh (hunk 2 above)
Comment (ds-ms) [reply images/linux/scripts/installers/docker.sh]: done

File: images/linux/scripts/installers/docker-moby.sh [hunk 3]
```diff
@@ -8,21 +8,20 @@
...
     apt-get update
     apt-get install -y apt-transport-https ca-certificates curl software-properties-common
-    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
-    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $LSB_CODENAME stable"
+    curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add -
+    curl https://packages.microsoft.com/config/ubuntu/16.04/prod.list > /etc/apt/sources.list.d/microsoft-prod.list
     apt-get update
-    apt-get install -y docker-ce
+    apt-get install -y moby-engine moby-cli
+    usermod -a -G docker $USER
```
Comment (dakale) [inline images/linux/scripts/installers/docker-moby.sh]: This runs in Packer as the `packer` user which is deleted at the end of provisioning, so this is not really needed

Comment (ds-ms) [reply images/linux/scripts/installers/docker-moby.sh]: Removed

**Pull request merged**